scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala

```diff
@@ -0,0 +1,87 @@
+package scala.meta.internal.parsers
+
+import scala.meta.Dialect
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token.Ident
+import scala.meta.internal.classifiers.classifier
+
+class SoftKeywords(dialect: Dialect) {
```

Comment:
do we need to wrap this in an instance? wouldn't this work, with implicit:
```
  object SoftKw {
    @classifier
    trait Using {
      def unapply(token: Token)(implicit dialect: Dialect): Boolean = {
        dialect.allowGivenUsing && isSoftKw(token, "using")
      }
    }
  }

  // somewhere else
  token.is[SoftKw.Using]
```

Reply:
Tried it and it does not recognize it as an unapply method :/